Build granularity changes (#201)
dfinity/docs: modules/developers-guide/pages/cli-reference.adoc

== dfx build

Use the `+dfx build+` command to compile your program into a WebAssembly module that can be deployed on the {IC}.
You can use this command to compile all of the programs that are defined for a project in the project's `+dfx.json+` configuration file or a specific canister.

Note that you can only run this command from within the project directory structure.
For example, if your project name is `+hello_world+`, your current working directory must be the `+hello_world+` top-level project directory or one of its subdirectories.

The `+dfx build+` command looks for the source code to compile using the information you have configured under the `+canisters+` section in the `+dfx.json+` configuration file.

=== Basic usage

[source,bash]
----

> dfx build [flag] [option] [--all | canister_name]

Reviewer: ```suggestion
dfx build [flag] [option] [--all | --canister-name]
```

|===

=== Arguments

You can specify the following arguments for the `+dfx build+` command.

[width="100%",cols="<36%,<64%",options="header"]
|===

|Argument |Description

|`+--all+` |Builds all of the canisters configured in the project's `+dfx.json+` file.

|`+canister_name+` |Specifies the name of the canister you want to build.

> If you are not using the `+--all+` option, the canister name is a required argument and must match at least one name that you have configured in the `+canisters+` section of the `+dfx.json+` configuration file for your project.

Reviewer: ```suggestion
If you are not using the `+--all+` option, you can continue to use `+dfx build+` or provide a canister name as an argument (the canister name must match at least one name that you have configured in the `+canisters+` section of the `+dfx.json+` configuration file for your project.)
```
